### PR TITLE
use gap 3 beta 2

### PR DESCRIPTION
**Pull request by stephanenicolas**
Extends #205. Update to GAP 3. 

TODO:
* [x] check dependencies are not exported too much (compiler related deps in transitive path)
* [x] snapshot is breaking java code version (52 cafe babe)

New Dependency Diagram:
image

### CODE REVIEW

File: smoothie/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest package="com.github.stephanenicolas.toothpick.smoothie">
+</manifest>
```
Comment (dlemures) [inline smoothie/src/main/AndroidManifest.xml]: empty line

**Comment by dlemures** [comment]
It looks OK.
Not very familiar with this though.

**Comment by stephanenicolas** [comment]
Do not merge yet.

File: smoothie-sample/build.gradle [hunk 2]
```diff
@@ -25,39 +22,49 @@ android {
...
   maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
 }
 
 dependencies {
-  compile project(':smoothie')
-  compile project(':toothpick-runtime')
+  implementation project(path: ':smoothie', configuration: 'default')
+  implementation project(':toothpick-runtime')
   // We cannot update to annotationProcessor for the sample as we need to keep compatibility with Java 7.
   // However, it is possible to use it instead of apt.
```
Comment (dlemures) [inline smoothie-sample/build.gradle]: the comment can be removed

File: smoothie-sample/build.gradle [hunk 3]
```diff
@@ -25,39 +22,49 @@ android {
...
-
-  testCompile project(':toothpick-testing')
-  testCompile deps.easymock
-  testCompile deps.truth
-  testCompile deps.robolectric
-}
+  annotationProcessor project(':toothpick-compiler')
+  annotationProcessor deps.butterknife_compiler
+  implementation deps.butterknife
+  implementation deps.rxandroid
```
Comment (dlemures) [inline smoothie-sample/build.gradle]: could we have the dependencies sorted by scope?

File: smoothie/build.gradle [hunk 4]
```diff
@@ -1,42 +1,51 @@
...
 
-apply plugin: 'java'
-apply plugin: 'org.inferred.processors'
+apply plugin: 'com.android.library'
 
-sourceCompatibility = 1.7
-targetCompatibility = 1.7
+dependencies {
+  api project(':toothpick')
+  api deps.inject
```
Comment (dlemures) [inline smoothie/build.gradle]: smoothie is an optional module including only bindings (and some providers used by them).

i think these 2 dependencies should be compileOnly too as `inject` and `project` should be already there

File: smoothie/gradle.properties [hunk 5]
```diff
@@ -1,3 +1,4 @@
 POM_ARTIFACT_ID=smoothie
 POM_NAME=smoothie
-POM_DESCRIPTION='Toothpick package providing Android providers'
\ No newline at end of file
+POM_DESCRIPTION='Toothpick package providing Android providers'
+POM_PACKAGING='aar'
```
Comment (dlemures) [inline smoothie/gradle.properties]: \n

File: toothpick-sample/build.gradle [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
 
 dependencies {
-  compile project(':toothpick')
-  compile project(':toothpick-runtime')
-  compile project(':toothpick-compiler')
-  compile deps.inject
+  implementation project(':toothpick')
+  implementation project(':toothpick-runtime')
+  annotationProcessor project(':toothpick-compiler')
+  implementation deps.inject
```
Comment (dlemures) [inline toothpick-sample/build.gradle]: sort be scope

File: toothpick-testing/build.gradle [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
+configurations {
+  testAnnotationProcessor
+}
+
 dependencies {
-  compile project(':toothpick-runtime')
-  compile deps.junit
+  api project(':toothpick-runtime')
+  api deps.junit
+  api deps.inject
```
Comment (dlemures) [inline toothpick-testing/build.gradle]: `project(':toothpick-runtime')` and `deps.inject` could be compileOnly?

Comment (dlemures) [reply toothpick-testing/build.gradle]: TP-testing is optional and TP should be already in PROD.

**Comment by stephanenicolas** [comment]
We need to update to beta 3 (and AS too)

**Comment by dlemures** [comment]
Updated scopes in new commit

**Pull request merged**